### PR TITLE
Add Tox environment for building the documentation.

### PR DESCRIPTION
**Pull request by wsanchez**
`tox -e docs` will generate `.tox/docs/html`

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -61,7 +61,6 @@ URLs have many parts, and URL objects have many attributes to represent them.
 .. autoattribute:: hyperlink.URL.userinfo
 .. autoattribute:: hyperlink.URL.user
 .. autoattribute:: hyperlink.URL.rooted
-.. autoattribute:: hyperlink.URL.family
```
Comment (wsanchez) [inline docs/api.rst]: Ain't such a thing as `URL.family`...

**Comment by mahmoud** [comment]
I think `parse_host` is deprecated and you're right, `parse` should definitely definitely get documented!

File: docs/api.rst [hunk 2]
```diff
@@ -71,5 +70,6 @@ A couple of notable helpers used by the :class:`~hyperlink.URL` type.
 .. autoclass:: hyperlink.URLParseError
 .. autofunction:: hyperlink.register_scheme
 .. autofunction:: hyperlink.parse_host
+.. autofunction:: hyperlink.parse
```
Comment (wsanchez) [inline docs/api.rst]: This one is public but not in the docs…

**Comment by wsanchez** [comment]
> I think `parse_host` is deprecated and you're right, `parse` should definitely definitely get documented!

OK I took `parse_host` back out of `__init__.py` and I removed it from the docs.

We might want to consider just removing that code, if it's not available except via `hyperlink._url`, which we wouldn't want to encourage people to import.  That said, that's a different PR.

**Comment by wsanchez** [comment]
@mahmoud This look OK?

**Comment by mahmoud** [comment]
@wsanchez hey sorry about the wait, yes, this looks mostly OK. I think it's a bit on the bleeding edge there with using py3.8, but I assume you've rendered and it looks ok? 

I've got some weird habits like using [sphinx-autobuild](https://pypi.org/project/sphinx-autobuild/), which I'm hoping continue to work (says it supports py3.6). As for those, I'll just deal with those bridges when I cross em.

**Comment by wsanchez** [comment]
I'll give `sphinx-autobuild` a spin.

**Comment by wsanchez** [comment]
> @wsanchez hey sorry about the wait, yes, this looks mostly OK. I think it's a bit on the bleeding edge there with using py3.8, but I assume you've rendered and it looks ok?

Looking at the rendered result… looks the same as on `hyperlink.readthedocs.io`.

**Comment by wsanchez** [comment]
> @wsanchez hey sorry about the wait, yes, this looks mostly OK. I think it's a bit on the bleeding edge there with using py3.8, but I assume you've rendered and it looks ok?
> 
> I've got some weird habits like using [sphinx-autobuild](https://pypi.org/project/sphinx-autobuild/), which I'm hoping continue to work (says it supports py3.6). As for those, I'll just deal with those bridges when I cross em.

Neat.  I added a `docs-auto` Tox environment, which will run `sphinx-autobuild` for you.

**Pull request merged**